crates/cargo-platform/src/cfg.rs

```diff
@@ -124,30 +145,40 @@ impl fmt::Display for Cfg {
         match *self {
             Cfg::Name(ref s) => s.fmt(f),
             Cfg::KeyPair(ref k, ref v) => write!(f, "{} = \"{}\"", k, v),
+            Cfg::Version(CfgRustVersion { minor, patch }) => {
+                write!(f, "version(\"1.{minor}.{patch}\")")
+            }
         }
     }
 }
 
 impl CfgExpr {
     /// Utility function to check if the key, "cfg(..)" matches the `target_cfg`
-    pub fn matches_key(key: &str, target_cfg: &[Cfg]) -> bool {
+    pub fn matches_key(key: &str, target_cfg: &[Cfg], rustc_version: &semver::Version) -> bool {
         if key.starts_with("cfg(") && key.ends_with(')') {
             let cfg = &key[4..key.len() - 1];
 
             CfgExpr::from_str(cfg)
                 .ok()
-                .map(|ce| ce.matches(target_cfg))
+                .map(|ce| ce.matches(target_cfg, rustc_version))
                 .unwrap_or(false)
         } else {
             false
         }
     }
 
-    pub fn matches(&self, cfg: &[Cfg]) -> bool {
+    pub fn matches(&self, cfg: &[Cfg], rustc_version: &semver::Version) -> bool {
         match *self {
-            CfgExpr::Not(ref e) => !e.matches(cfg),
-            CfgExpr::All(ref e) => e.iter().all(|e| e.matches(cfg)),
-            CfgExpr::Any(ref e) => e.iter().any(|e| e.matches(cfg)),
+            CfgExpr::Not(ref e) => !e.matches(cfg, rustc_version),
+            CfgExpr::All(ref e) => e.iter().all(|e| e.matches(cfg, rustc_version)),
+            CfgExpr::Any(ref e) => e.iter().any(|e| e.matches(cfg, rustc_version)),
+            CfgExpr::Value(Cfg::Version(CfgRustVersion { minor, patch })) => {
+                match minor.cmp(&rustc_version.minor) {
+                    Ordering::Less => true,
+                    Ordering::Equal => patch <= rustc_version.patch,
+                    Ordering::Greater => false,
+                }
+            }
```

Comment:
Like we did with `package.rust-version`, could we delegate the matching logic to `CfgRustVersion`?